toutoumomoma.go

```diff
@@ -177,31 +178,12 @@ func (f *File) Stripped() (sneaky bool, err error) {
 //	kernel32.writeconsolew
 //	kernel32.waitformultipleobjects
 func (f *File) ImportHash() (hash []byte, imports []string, err error) {
-	// Algorithm from https://www.fireeye.com/blog/threat-research/2014/01/tracking-malware-import-hashing.html
-	//  - Resolving ordinals to function names when they appear (done by the debug/pe library)
-	//  - Converting both DLL names and function names to all lowercase
-	//  - Removing the file extensions from imported module names
-	//  - Building and storing the lowercased string in an ordered list
-	//  - Generating the MD5 hash of the ordered list
-	//
-	// The algorithm is generalised to non-Windows platforms as described in
-	// the doc comment.
+	return f.importHash()
+}
 
-	imports, err = f.importedSymbols()
-	if err != nil {
-		return nil, nil, err
-	}
-	h := md5.New()
-	if len(imports) == 0 {
-		return h.Sum(nil), nil, nil
-	}
-	for i, imp := range imports {
-		if i != 0 {
-			_, _ = h.Write([]byte{','})
-		}
-		fmt.Fprint(h, imp)
-	}
-	return h.Sum(nil), imports, nil
+// Imports returns the list of dynamic imports in the executable examined.
+func (f *File) Imports() (imports []string, err error) {
```

Comment:
What is the need for introducing a new public method here and below? I don't see a clear, connected usecase and therefore would keep this separated from fips specific changes.

Reply:
Ah so you are going to update auditbeat to call `f.Imports()` when `f.ImportHash()` fails. Ok, this makes sense.